packages/main/src/plugin/extension-loader.ts

```diff
@@ -1091,6 +1132,18 @@ export class ExtensionLoader {
       .catch((error: unknown) => console.log(`Failed to track deactivateExtension telemetry event. Error: ${error}`));
   }
 
+  async stopExtension(extensionId: string): Promise<void> {
+    // the user explicitly disabled the extension, save the configuration
+    const disabledExtensions = this.getDisabledExtensionIds();
+
+    if (!disabledExtensions.includes(extensionId)) {
+      disabledExtensions.push(extensionId);
+      this.setDisabledExtensionIds(disabledExtensions);
+    }
+
```

Comment:
here it would be easier if disabled is already stored in the extension.

it's strange also that we don't check if extension exists before adding it to the list ?

Reply:
I'm not sure to understand also what it does.

if I stop and extension and the extension is not in the disabled list, it adds this extension as being disabled

so it is not a stop but a disableExtension method